cmake/check_linker_flag.cmake

```diff
@@ -0,0 +1,53 @@
+include(CheckCXXSourceCompiles)
+# We need some extra FAIL_REGEX patterns
+# Note that CHECK_C_SOURCE_COMPILES is a misnomer, it will also link.
+SET(fail_patterns
+    FAIL_REGEX "argument unused during compilation"
+    FAIL_REGEX "unsupported .*option"
+    FAIL_REGEX "unknown .*option"
+    FAIL_REGEX "unrecognized .*option"
+    FAIL_REGEX "ignoring unknown option"
+    FAIL_REGEX "warning:.*ignored"
+    FAIL_REGEX "warning:.*is valid for.*but not for"
+    FAIL_REGEX "warning:.*redefined"
+    FAIL_REGEX "[Ww]arning: [Oo]ption"
+    )
+#The regex patterns above are not localized, thus LANG=C
+SET(ENV{LANG} C)
+
+MACRO (MY_CHECK_MODULE_LINKER_FLAG flag)
```

Comment:
Why "MODULE", How does it check specifically for MODULE?

Reply:
This was the WIP bit, waiting for [a clue](https://stackoverflow.com/questions/61604663/how-do-i-test-cmake-shared-linker-flags-and-cmake-module-linker-flags) to see how to test module vs shared library vs exe . I couldn't immediately see one, so was keeping it separate for now. If your happy with just this implementation I can rename/merge it with MY_CHECK_LINKER_FLAG.